### PR TITLE
cfssl api test suite

### PR DESCRIPTION
**Pull request by anniephan**
Made new PR due to some technical difficulties with the last branch.

### CODE REVIEW

**Comment by anniephan** [comment]
Travis seems to be failing from extraneous tests...

File: cmd/cfssl/cfssl_test.go [hunk 1]
```diff
@@ -1 +1,421 @@
...
+	return apiRes
+}
+
+func responseUnmarshal(body []byte) api.Response {
+	res := api.Response{}
+	json.Unmarshal(body, &res)
+	return res
+}
+
+/* START OF TESTS*/
```
Comment (grittygrease) [inline cmd/cfssl/cfssl_test.go]: C-style comments are usually reserved for large blocks of text or code, see https://golang.org/doc/effective_go.html

Comment (grittygrease) [inline cmd/cfssl/cfssl_test.go]: Most of the function comments here can be C++ style.

File: cmd/cfssl/cfssl_test.go [hunk 2]
```diff
@@ -1 +1,421 @@
...
+		t.Fatalf("NewKey API Failed Response")
+	}
+}
+
+func TestAuthSign(t *testing.T) {
+	jsonStr := []byte(`{"token": "", "request": ` + string(certRequest) + `}`)
+	response := apiOut("POST", jsonStr, postStr+"authsign", t)
+	signAPIResp := apiFormat(response)
+	if !signAPIResp.Success {
+		//t.Fatalf("AuthSign API Failed Response")
```
Comment (grittygrease) [inline cmd/cfssl/cfssl_test.go]: Intentional?

File: cmd/cfssl/cfssl_test.go (hunk 1 above)
Comment (anniephan) [inline cmd/cfssl/cfssl_test.go]: Ah I see, thanks. I'll fix up the comment styling with the next commit.

**Pull request closed without merging**